crates/red_knot_python_semantic/src/semantic_index/use_def/symbol_state.rs

```diff
@@ -314,7 +314,7 @@ impl SymbolBindings {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub(super) struct SymbolState {
+pub(in crate::semantic_index) struct SymbolState {
```

Comment:
:exploding_head: We _do_ have `pub(grandparent)` as a possibility, I did not know that! I still worry that we might be over-thinking visibility within this crate, but that is not at all a blocker for this PR.